web/js/main_search.js

```diff
@@ -83,9 +83,36 @@ $(document).ready(function() {
 
     // Switch source and target locales in select boxes
     $('#locale_switch').on('click', function() {
-      source_locale = $('#source_locale').val()
-      target_locale = $('#target_locale').val()
-      $('#source_locale').val(target_locale).change();
-      $('#target_locale').val(source_locale).change();
+        source_locale = $('#source_locale').val()
+        target_locale = $('#target_locale').val()
+        $('#source_locale').val(target_locale).change();
+        $('#target_locale').val(source_locale).change();
     });
+
+    // Hide the clear button if the search field is empty
+    if (! $('#recherche').val()) {
+        $('#clear').hide();
+    } else {
+        $('#clear').show();
+    }
+
+    // Show the clear button if search field changes
+    $('#recherche').on('input',function() {
+         if ($(this).data('lastval') != this.value){
+            $(this).data('lastval', this.value);
+            $('#clear').show();
+        };
+
+        // If we delete the text with backspace, hide the button
+        if (! this.value) {
+            $('#clear').hide();
+        };
+    });
+
```

Comment:
Unless I'm missing something, you should be able to replace line 92-110 with
```
    // Hide the clear button if the search field is empty
    $('#recherche').on('input', function() {
        if (!this.value) {
            $('#clear').hide();
        } else {
            $('#clear').show();
        }
    });
```
And adding ```display: none``` in the CSS for `#clear` (when loading the page the field is empty).

Side note: we're not using a space after ! in JS, also don't need the ; after an `if`.